### PR TITLE
feat(SearchBar): Migrate SearchBar from Products v1 / Security to Products v2

### PR DESCRIPTION
**Pull request by makafsal**
Contributes to #4149 

This is to migrate the SearchBar component to the main

#### What did you change?
Implemented a new SearchBar component, stories, and tests

#### How did you test and verify your work?
React testing library
Storybook

- [x] Design review @sdignum

### CODE REVIEW

File: packages/ibm-products/src/components/SearchBar/SearchBar.test.js [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+    const searchBox = screen.getByDisplayValue(value);
+    const submitButton = screen.getByText(defaultProps.submitLabel);
+    const clearButton = screen.getByTitle(defaultProps.clearButtonLabelText);
+
+    expect(clearButton).toBeInTheDocument();
+    expect(searchBox).toBeInTheDocument();
+    expect(submitButton).toBeEnabled();
+
+    fireEvent.click(submitButton);
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/SearchBar/SearchBar.test.js:81]: We've been trying to only use `userEvent` for interactions, ie:
```js
const { click } = userEvent;
await click(submitButton);
```

File: packages/ibm-products/src/components/SearchBar/SearchBar.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+// Import portions of React that are needed.
+import React, { useEffect, useState } from 'react';
+
+// Other standard imports.
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import { getDevtoolsProps } from '../../global/js/utils/devtools';
+import { pkg /*, carbon */ } from '../../settings';
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/SearchBar/SearchBar.js:16]: Remove `carbon` import since it's not being used

**Comment by sdignum** [comment]
Looks good to me. Design approved! @elycheea @makafsal

File: packages/ibm-products/src/components/SearchBar/SearchBar.test.js (hunk 1 above)
Comment (makafsal) [reply packages/ibm-products/src/components/SearchBar/SearchBar.test.js]: Resolved.

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**